ext/cl_khr_tensor.asciidoc

```diff
@@ -0,0 +1,547 @@
+  being queried is returned. If _param_value_ is NULL, it is ignored.
+
+* _param_value_size_ is used to specify the size in bytes of memory
+  pointed to by _param_value_. This size must be ≥ size of return type
+  as described in the <<Tensor Object Queries>> table.
+
+* _param_value_size_ret_ returns the actual size in bytes of data
+  being queried by _param_name_. If _param_value_size_ret_ is NULL, it is
+  ignored.
+
+*clGetTensorInfo* returns CL_SUCCESS if the function is executed
+ succesfully. Otherwise, it returns one of the following errors:
+
+* CL_INVALID_TENSOR if _tensor_ is not a valid tensor object.
+
+[#Tensor Object Quaries]
+.List of supported param_names by clGetTensorInfo
+[cols="2,1,2",stripes=odd]
+|===
+| CL_TENSOR_RANK  | size_t         | Return the tensor rank.
+| CL_TENSOR_SHAPE | size_t[]       | Return the tensor shape.
+| CL_TENSOR_DTYPE | cl_tensor_type | Return the tensor data type.
+
+| CL_TENSOR_COMMAND_BUFFER_TEMPORARY | cl_bool | Return true if the
+tensor is temporary tensor for command buffers.
+
+| CL_TENSOR_BOUND_TO_BUFFER | cl_bool | Return true if the tensor is
+bound to a buffer. If CL_TENSOR_COMMAND_BUFFER_TEMPORARY is true, then
+CL_TENSOR_BOUND_TO_BUFFER must return false.
+
+| CL_TENSOR_BUFFER | cl_mem a| If CL_TENSOR_BOUND_TO_BUFFER is true,
+return the buffer object the tensor is bound to. Otherwise,
+clGetTensorInfo call returns:
+
+* CL_INVALID_MEM_OBJECT if the tensor is not bound to a buffer object.
+
+* CL_INVALID_PROPERTY otherwise.
+
+| CL_TENSOR_CONTEXT | cl_context | Return the context specified when
+  the tensor object is created.
+
+| CL_TENSOR_REFERENCE_COUNT | cl_uint | Return the tensor reference
+count.
+|===
+
+To read from a tensor to host memory / buffer object or to write to a
+tensor object from host memory / buffer object call one of the functions.
+
+[source,c]
+----
+cl_int clEnqueueReadTensor(
+  cl_command_queue command_queue,
+  cl_tensor tensor,
+  cl_bool blocking_command,
+  cl_mem buffer,
+  void* host_ptr,
+  cl_uint num_events_in_wait_list,
+  const cl_event* event_wait_list,
+  cl_event* event);
+----
+
+[source,c]
+----
+cl_int clEnqueueWriteTensor(
+  cl_command_queue command_queue,
+  cl_tensor tensor,
+  cl_bool blocking_command,
+  cl_mem buffer,
+  void* host_ptr,
+  cl_uint num_events_in_wait_list,
+  const cl_event* event_wait_list,
+  cl_event* event);
+----
+
+* _command_queue_ is a valid host command-queue in which the read /
+  write command will be queued. _command_queue_ and _tensor_ must be
+  created with the same OpenCL context.
+
+* _tensor_ refers to a valid tensor object which is bound to a buffer.
+
+* _blocking_command_ indicate if the read and write operations are
+  blocking or non-blocking (see below).
+
+* _buffer_ refers to a valid buffer object where data is to be
+  read into or to be written from when the value of _host_ptr_ is
+  NULL. If _host_ptr_ is non-NULL then value of _buffer_ is ignored.
```

Comment:
It might be less error-prone to require the buffer to be NULL when the host pointer is non-NULL.

Reply:
This should be now reflected in the extension.